### ISSUE TITLE
pub serve gets stuck, never serving files

### ISSUE THREAD

**Issue by jelbourn** [angular/angular#2628]
Trying to get a component working, I found that having an html file that contains a non-empty `<template>` element caused `pub serve` to seemingly get stuck in an infinite loop (and never serve any requests). 

@yjbanov suggested this had something to do with the dart `Html5LibDomAdapter`'s `appendChild` shim returning `null` instead of throwing a not implemented exception. Changing this shim worked insofar as to make my component work, but now there is an exception logged:

```
Could not find Directive entry for formDirectives
<html template>
<html span>
<html template>
[Error from EmptyNgDepsRemover on examples|web/src/hello_world/index_dynamic.ng_deps.dart]:<html span>

Parsing ng templates failed.
Exception: not implemented
Stack Trace: #0      Html5LibDomAdapter.appendChild (package:angular2/src/dom/html_adapter.dart:132:5)
#1      ViewSplitter._addParentElement (package:angular2/src/render/dom/compiler/view_splitter.dart:103:9)
#2      ViewSplitter.process (package:angular2/src/render/dom/compiler/view_splitter.dart:88:14)
#3      CompileControl.internalProcess (package:angular2/src/render/dom/compiler/compile_control.dart:33:12)
#4      CompilePipeline._process (package:angular2/src/render/dom/compiler/compile_pipeline.dart:39:23)
#5      CompilePipeline._process (package:angular2/src/render/dom/compiler/compile_pipeline.dart:55:16)
#6      CompilePipeline.process (package:angular2/src/render/dom/compiler/compile_pipeline.dart:32:10)
#7      _TemplateExtractor.extractTemplates.<extractTemplates_async_body> (package:angular2/src/transform/template_compiler/generator.dart:107:18)
#8      _RootZone.runUnary (dart:async/zone.dart:1155)
#9      _Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:494)
#10     _Future._propagateToListeners (dart:async/future_impl.dart:577)
#11     _Future._completeWithValue (dart:async/future_impl.dart:368)
#12     Future.wait.<anonymous closure> (dart:async/future.dart:299)
#13     _RootZone.runUnary (dart:async/zone.dart:1155)
#14     _Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:494)
#15     _Future._propagateToListeners (dart:async/future_impl.dart:577)
#16     _Future._completeWithValue (dart:async/future_impl.dart:368)
#17     _Future._asyncComplete.<anonymous closure> (dart:async/future_impl.dart:422)
#18     _asyncRunCallbackLoop (dart:async/schedule_microtask.dart:41)
#19     _asyncRunCallback (dart:async/schedule_microtask.dart:48)
#20     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:96)
#21     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:143)
```
This error is logged 3 times.

**Comment by tbosch** [comment]
Related to #2627

**Comment by tbosch** [comment]
Fixed by #2627
